pkg/actions/service_test.go

```diff
@@ -33,108 +44,44 @@ func NewActionStatsMockCollector() ActionStatsMockCollector {
-		EventType:        graveler.EventTypePreCommit,
-		StorageNamespace: "storageNamespace",
-		RepositoryID:     "repoID",
-		BranchID:         "branchID",
-		SourceRef:        "sourceRef",
-		Commit: graveler.Commit{
-			Message:   "commitMessage",
-			Committer: "committer",
-			Metadata:  map[string]string{"key": "value"},
-		},
-	}
-	const actionName = "test action"
-	const webhookID = "webhook_id"
-	const airflowHookID = "airflow_hook_id"
-	const airflowHookIDNoConf = "airflow_hook_id_no_conf"
-	hookResponse := "OK"
-
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer func() { _ = r.Body.Close() }()
-		data, err := io.ReadAll(r.Body)
-		if err != nil {
-			t.Error("Failed to read webhook post data", err)
-			return
-		}
-
-		if r.URL.Path == "/webhook" {
-			queryParams := map[string][]string(r.URL.Query())
-			require.Len(t, queryParams["prefix"], 1)
-			require.Equal(t, "public/", queryParams["prefix"][0])
-			require.Len(t, queryParams["disallow"], 2)
-			require.Equal(t, "user_", queryParams["disallow"][0])
-			require.Equal(t, "private_", queryParams["disallow"][1])
-
-			var eventInfo actions.EventInfo
-			err = json.Unmarshal(data, &eventInfo)
-			if err != nil {
-				t.Error("Failed to unmarshal webhook data", err)
-				return
-			}
+func (c *ActionStatsMockCollector) CollectMetadata(_ *stats.Metadata) {}
+func (c *ActionStatsMockCollector) SetInstallationID(_ string)        {}
+func (c *ActionStatsMockCollector) Close()                            {}
 
-			checkEvent(t, record, eventInfo, actionName, webhookID)
-		} else if r.URL.Path == "/airflow/api/v1/dags/some_dag_id/dagRuns" {
-			var req actions.DagRunReq
+type getService func(t *testing.T, ctx context.Context, source actions.Source, writer actions.OutputWriter, stats stats.Collector, runHooks bool) actions.Service
 
-			withConf := r.URL.Query().Get("conf") == "true"
-			expectedID := airflowHookID
-			if !withConf {
-				expectedID = airflowHookIDNoConf
-			}
-
-			require.NoError(t, json.Unmarshal(data, &req))
-			require.True(t, strings.HasPrefix(req.DagRunID, "lakeFS_hook_"+expectedID))
-
-			if withConf {
-				require.Equal(t, req.Conf["some"], "additional_conf")
-			}
+func GetDBService(t *testing.T, ctx context.Context, source actions.Source, writer actions.OutputWriter, stats stats.Collector, runHooks bool) actions.Service {
+	t.Helper()
+	conn, _ := testutil.GetDB(t, databaseURI)
+	return actions.NewDBService(ctx, conn, source, writer, stats, runHooks)
+}
 
-			username, pass, ok := r.BasicAuth()
-			require.True(t, ok)
-			require.Equal(t, "some_username", username)
-			require.Equal(t, "some_password", pass)
+func GetKVService(t *testing.T, ctx context.Context, source actions.Source, writer actions.OutputWriter, stats stats.Collector, runHooks bool) actions.Service {
+	t.Helper()
+	kvStore := kvtest.GetStore(ctx, t)
+	return actions.NewService(ctx, kv.StoreMessage{Store: kvStore}, source, writer, stats, runHooks)
+}
```

Comment:
Great stuff!! If only Github would know to recognize that indentations are not a change..